Extend documentation for Hive metadata tables (#18141)
trinodb/trino: docs/src/main/sphinx/connector/hive.rst

The raw Hive table properties are available as a hidden table, containing a
separate column per table property, with a single row containing the property
values.

``$properties`` table
~~~~~~~~~~~~~~~~~~~~~

> The properties table name is the same as the table name with ``$properties`` appended

Reviewer: ```suggestion
The properties table name is composed with the table name and ``$properties`` appended.
```

> and exposes the parameters of the table in the metastore.

Reviewer: ```suggestion
It exposes the table properties stored in the metastore.
```

You can inspect the property names and values with a simple query::

    SELECT * FROM example.web."page_views$properties";

.. code-block:: text

>             stats_generated_via_stats_task        | auto.purge |       presto_query_id       | presto_version | transactional

Reviewer: presto_version ???

Reviewer: And presto_query_id ?

Reviewer: sigh .. I guess thats a backwards compatibility thing

     ---------------------------------------------+------------+-----------------------------+----------------+---------------
      workaround for potential lack of HIVE-12730 | false      | 20230705_152456_00001_nfugi | 423            | false

``$partitions`` table
~~~~~~~~~~~~~~~~~~~~~

> The ``$partitions`` table provides the entire listing of the partition values

Reviewer: ```suggestion
The ``$partitions`` table provides a list of all partition values
```

> of a partitioned table.

Reviewer: ```suggestion
of a partitioned table::
```

> from the partitioned table.

Reviewer: ```suggestion
of a the partitioned table.
```

> You can inspect the partition values with a simple query::

Reviewer: ```
The following example query returns all partition values from the ``page_views`` table in the ``web`` schema of the ``example`` catalog::
```
